kong/db/strategies/postgres/connector.lua

```diff
@@ -215,25 +222,18 @@ local _mt = {}
 _mt.__index = _mt
 
 
-local function extract_major_minor(release_version)
-  return string.match(release_version, "^(%d+%.%d+)")
-end
-
-
 function _mt:init()
-  local res, err = self:query("SHOW server_version;")
-  if not res then
+  local res, err = self:query("SHOW server_version_num;")
+  local ver = tonumber(res and res[1] and res[1].server_version_num)
+  if not ver then
     return nil, "failed to retrieve server_version: " .. err
   end
 
-  if #res < 1 or not res[1].server_version then
-    return nil, "failed to retrieve server_version"
-  end
-
-  self.major_minor_version = extract_major_minor(res[1].server_version)
-  if not self.major_minor_version then
-    return nil, "failed to extract major.minor version from '" ..
-                res[1].server_version .. "'"
+  local major = floor(ver / 10000)
+  if major < 10 then
+    self.major_minor_version = fmt("%u.%u.%u", major, floor(ver / 100 % 100), ver % 100)
```

Comment:
`major_minor` is supposed to only contain `X.Y` from the `X.Y.Z` version string, on purpose. It looks like this would break this expectation? The old DAO has tests for this for both C* and Postgres (that were not yet ported to the new DAO): https://github.com/Kong/kong/blob/next/spec/01-unit/01-db/03-postgres_spec.lua

Reply:
So question is, should we follow meaning of Major/Minor of PG, or not, and follow semver style, even when it doesn't match the real world.

Reply:
And I can port the tests to new db too.

Reply:
I have pushed changes to this branch.